openstates/la/legislators.py

```diff
@@ -24,66 +20,63 @@ class LALegislatorScraper(LegislatorScraper, BackoffScraper, LXMLMixin):
 
     def scrape_upper_leg_page(self, term, url, who):
         page = self.lxmlize(url)
-        info = page.xpath("//td[@bgcolor='#EBEAEC']")
-        who = page.xpath("//font[@size='4']")
-        who = who[0].text_content()
-        who = re.sub("\s+", " ", who)
-        who, district = (x.strip() for x in who.rsplit("-", 1))
-        who = who.replace("Senator", "").strip()
-        district = district.replace("District", "").strip()
-
-        infopane = page.xpath("//table[@cellpadding='3']")
-        infos = [x.tail.strip() if x.tail else ""
-                 for x in infopane[1].xpath(".//br")]
-
-        keys = ["party", "email", "capitol-office",
-                "district-office", "phone", "fax", "staffer"]
-        nodes = [[]]
-        for node in infos:
-            if node == "":
-                if nodes[-1] != []:
-                    nodes.append([])
-                continue
-            nodes[-1].append(node)
-
-        data = dict(zip(keys, nodes))
 
-        district_office = "\n".join(data['district-office'])
-        capitol_office = "\n".join(data['capitol-office'])
+        (who, ) = [x for x in
+                   page.xpath('//tr/td/font/text()') if
+                   x.strip().startswith("Senator ")
+                   ]
+        who = re.search(r'(?u)^\s*Senator\s*(.*?)\s*$', who).group(1)
+
+        (district, ) = [x for x in
+                        page.xpath('//tr/td/font/text()') if
+                        x.strip().startswith("District - ")
+                        ]
+        district = re.search(
+            r'(?u)^\s*District\s*-\s*(.*?)\s*$', district).group(1)
+
+        info = [x.strip() for x in
+                page.xpath('//font[contains(text(), "Information:")]/'
+                'ancestor::table[1]//text()') if
+                x.strip()
+                ]
 
-        rundown = infopane[1].xpath("./*")[-1]
-        rundown_txt = rundown.text_content()
         parties = {
             "Republican": "Republican",
             "Democrat": "Democratic",
         }
+        party_index = info.index("Party:") + 1
+        party = parties[info[party_index]]
 
-        party = 'other'
-        for slug in parties:
-            if slug in rundown_txt:
-                party = parties[slug]
+        phone_index = info.index("District Phone") + 1
+        phone = info[phone_index]
+        assert (sum(c.isdigit() for c in phone) == 9,
+                "Phone number is invalid: {}".format(phone))
```

Comment:
because of the way the python parer works -- this will always evaluate to `True`, hilariously

(Since it'll turn it into a tuple, and evalue that as a non-empty typle, making it truthy -- rather then parsing that as the second argument - the joys of statements in Python)

This assertion will also fail for a few folks :)

Let's talk this stuff over on Monday!